java-checks/src/main/java/org/sonar/java/checks/spring/SpelExpressionCheck.java

```diff
@@ -0,0 +1,301 @@
+  }
+
+  @CheckForNull
+  private static Map.Entry<Tree, String> extractArgumentValue(ExpressionTree expression) {
+    expression = getExpressionOrAssignmentRhs(expression);
+    var stringValue = ExpressionsHelper.getConstantValueAsString(expression).value();
+    if (stringValue == null) {
+      return null;
+    }
+    return Map.entry(expression, stringValue);
+  }
+
+  private static ExpressionTree getExpressionOrAssignmentRhs(ExpressionTree expression) {
+    return expression.is(Tree.Kind.ASSIGNMENT) ? ((AssignmentExpressionTree) expression).expression() : expression;
+  }
+
+  private void checkSpringExpressionsInString(Map.Entry<Tree, String> entry) {
+    var expression = entry.getKey();
+    try {
+      var argValue = entry.getValue();
+      if (expression.is(Tree.Kind.STRING_LITERAL)) {
+        checkStringContents(argValue, 1);
+      } else {
+        checkStringContents(argValue, 0);
+      }
+    } catch (SyntaxError e) {
+      reportIssue(expression, e);
+    }
+  }
+
+  private void reportIssue(Tree expression, SyntaxError error) {
+    if (expression.is(Tree.Kind.STRING_LITERAL)) {
+      // For string literals, report exact issue location within the string.
+      var tokenStart = Position.startOf(expression);
+      var textSpan = new AnalyzerMessage.TextSpan(
+        tokenStart.line(),
+        tokenStart.columnOffset() + error.startColumn,
+        tokenStart.line(),
+        tokenStart.columnOffset() + error.endColumn
+      );
+
+      var analyzerMessage = new AnalyzerMessage(this, context.getInputFile(), textSpan, error.getMessage(), 0);
+      ((DefaultJavaFileScannerContext) context).reportIssue(analyzerMessage);
+    } else {
+      reportIssue(expression, error.getMessage());
+    }
+  }
+
+  private static void checkStringContents(String content, int startColumn) throws SyntaxError {
+    var i = 0;
+    while (i < content.length()) {
+      var c = content.charAt(i);
+      switch (c) {
+        case '$':
+          i = parseDelimitersAndContents(content, i + 1, startColumn + i, SpelExpressionCheck::parseValidPropertyPlaceholder);
+          break;
+        case '#':
+          i = parseDelimitersAndContents(content, i + 1, startColumn + i, SpelExpressionCheck::parseValidSpelExpression);
+          break;
+        default:
+          i++;
+          break;
+      }
+    }
+  }
+
+  /**
+   * Parses the following grammatical expression, starting at <code>startIndex</code> in `value`:
+   *
+   * <pre>
+   * ('{' contents '}')?
+   * </pre>
+   * <p>
+   * Where correct bracing is checked and then <code>contents</code> is parsed using the given <code>parseContents</code> function.
+   *
+   * @param value         string containing the character sequence to parse
+   * @param startIndex    start index of the character sequence in <code>value</code>
+   * @param startColumn   column offset between the original string literal and <code>value</code> (needed for reporting)
```

Comment:
Thank you for the change, my point was also about understanding why these two values would be different.
I was expecting something closer to
```suggestion
   * @param startIndex       index of the opening delimiter we start from in the <code>value</code> value
   * @param startColumn   offset with the position of  <code>value</code> within a potentially longer original string (used for reporting)
```

Reply:
Let's be careful when mixing index and column terms because index is generic and 0-based, while column could be considered analyzer jargon and lead to confusion